### PR TITLE
Customize opensearch deployment annotation through values.yaml

### PR DESCRIPTION
**Pull request by prathaptce**
### Description
[Currently, OpenSearch deployment is not customizable using values.yaml . I'm trying to add deployment annotation in the values.yaml , which will be used by the deployment.yaml and configures the metadata annotation of the deployment.]
 
### Issues Resolved
[https://github.com/opensearch-project/opensearch-devops/issues/115]
 
### Check List
- [x] Commits are signed per the DCO using --signoff

For any changes to files within Helm chart directories:
- [x] Helm chart version bumped
- [x] Helm chart `CHANGELOG.md` updated to reflect change

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/helm-charts/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by prathaptce — COMMENTED** [review]
Currently, OpenSearch deployment is not customizable using values.yaml . I'm trying to add deployment annotation in the values.yaml , which will be used by the deployment.yaml and configures the metadata annotation of the deployment.

Use case: In the DR site deployments, we want to control the deployment of OpenSearch based on a few Deployment annotations. which is not possible now as deployment annotations cannot be passed through values.yaml

**Comment by prudhvigodithi** [comment]
Hey @prathaptce thanks for the contribution, please check the following points.
1) The DCO check is missing, please make sure the commits are signed.
2) The Helm chart version has to be bumped (sample [PR](https://github.com/opensearch-project/helm-charts/pull/376/files)).
3) The `CHANGELOG.md` has to be recorded along with if required updating `README.md`.

Apart from this I will review the technical aspects of the chart shortly, thank you.

**Comment by prathaptce** [comment]
> Hey @prathaptce thanks for the contribution, please check the following points.
> 
> 1. The DCO check is missing, please make sure the commits are signed.
> 2. The Helm chart version has to be bumped (sample [PR](https://github.com/opensearch-project/helm-charts/pull/376/files)).
> 3. The `CHANGELOG.md` has to be recorded along with if required updating `README.md`.
> 
> Apart from this I will review the technical aspects of the chart shortly, thank you.

Now, DCO check is passing, helm chart version is bumped & 'CHANGELOG.md' is updated with the changes done for this PR

File: charts/opensearch/values.yaml [hunk 1]
```diff
@@ -131,6 +131,9 @@ image:
 
 podAnnotations: {}
   # iam.amazonaws.com/role: es-cluster
+  
+# Deployment annotations
+annotations: {}
```
Comment (prudhvigodithi) [inline charts/opensearch/values.yaml:136]: Hey Can we please use as `openSearchAnnotations`?, also please change the comment to `# OpenSearch Statefulset annotations`
Thank you

File: charts/opensearch-dashboards/values.yaml [hunk 2]
```diff
@@ -67,6 +67,9 @@ secretMounts: []
 
 podAnnotations: {}
 
+# Deployment annotations
+annotations: {}
```
Comment (prudhvigodithi) [inline charts/opensearch-dashboards/values.yaml:71]: Hey Can we please use as `dashboardAnnotations`?

**Comment by prudhvigodithi** [comment]
Hey @prathaptce apart from my previous suggestions, can you take care of the lint errors also please update the README files in the chart with the new annotation changes.
Thank you

File: charts/opensearch-dashboards/templates/deployment.yaml [hunk 3]
```diff
@@ -3,6 +3,10 @@ kind: Deployment
 metadata:
   name: {{ template "opensearch-dashboards.fullname" . }}
   labels: {{- include "opensearch-dashboards.labels" . | nindent 4 }}
+  {{- with .Values.annotations }}
```
Comment (prudhvigodithi) [inline charts/opensearch-dashboards/templates/deployment.yaml:6]: Would be as `dashboardAnnotations`.

File: charts/opensearch/values.yaml (hunk 1 above)
Comment (prathaptce) [reply charts/opensearch/values.yaml]: Changed to openSearchAnnotations & comments to stateful set annotations

File: charts/opensearch-dashboards/values.yaml (hunk 2 above)
Comment (prathaptce) [reply charts/opensearch-dashboards/values.yaml]: Addressed this comment as well

File: charts/opensearch-dashboards/templates/deployment.yaml (hunk 3 above)
Comment (prathaptce) [reply charts/opensearch-dashboards/templates/deployment.yaml]: made the necessary changes here as well

File: charts/opensearch/templates/statefulset.yaml [hunk 4]
```diff
@@ -7,6 +7,9 @@ metadata:
     {{- include "opensearch.labels" . | nindent 4 }}
   annotations:
     majorVersion: "{{ include "opensearch.majorVersion" . }}"
+    {{- with .Values.annotations }}
```
[the remark this answers is not in the document]
Comment (prathaptce) [reply charts/opensearch/templates/statefulset.yaml]: incorporated the changes

**Comment by prathaptce** [comment]
> Hey @prathaptce apart from my previous suggestions, can you take care of the lint errors also please update the README files in the chart with the new annotation changes. Thank you

I'm not able to execute ct lint locally, is there a way I can trigger the ct lint job in GitHub & check the error? I'm not sure which line is creating the lint error

File: charts/opensearch-dashboards/CHANGELOG.md [hunk 5]
```diff
@@ -13,6 +13,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Fixed
 ### Security
 ---
+## [2.9.2]
+### Added
+- Added custom opensearch deployment annotation through values.yaml
```
Comment (prudhvigodithi) [inline charts/opensearch-dashboards/CHANGELOG.md:18]: [NIT] Thanks @prathaptce noticed one more, please check
- Added custom opensearch and dashboard annotations through values.yaml.

Comment (prathaptce) [reply charts/opensearch-dashboards/CHANGELOG.md]: thanks. updated the same

**Comment by prudhvigodithi** [comment]
Thanks @prathaptce , @TheAlgo and @peterzhuamazon can you please review ?
Thank you

**Comment by prathaptce** [comment]
> Thanks @prathaptce , @TheAlgo and @peterzhuamazon can you please review ? Thank you

@TheAlgo @peterzhuamazon, can you please review? thank you

**Comment by prudhvigodithi** [comment]
Hey @prathaptce I did notice one thing the `Chary.yaml` file is not incremented, sample PR https://github.com/opensearch-project/helm-charts/pull/375/files, can you please check.

**Comment by prathaptce** [comment]
> Hey @prathaptce I did notice one thing the `Chary.yaml` file is not incremented, sample PR https://github.com/opensearch-project/helm-charts/pull/375/files, can you please check.

Just now pumped up the version numbers in Chart.yaml. Please check now

**Review by prudhvigodithi — APPROVED** [review]

**Comment by prudhvigodithi** [comment]
Pinging @TheAlgo and @peterzhuamazon

**Comment by prathaptce** [comment]
hi @TheAlgo @peterzhuamazon, Can you please review the changes & let me know your comments?

**Comment by prathaptce** [comment]
Can this PR merged? if no other comments. thanks

**Comment by prudhvigodithi** [comment]
Hey @TheAlgo @peterzhuamazon please add your approval. Thanks for your patience @prathaptce 
Thank you

**Comment by TheAlgo** [comment]
> Hey @TheAlgo @peterzhuamazon please add your approval. Thanks for your patience @prathaptce Thank you

Will check this week. Apologies for delays.

**Review by TheAlgo — APPROVED** [review]
LGTM!

**Comment by prathaptce** [comment]
@TheAlgo .Thank you. Can you please merge this PR?

**Comment by bbarani** [comment]
Merging the PR as its approved.

**Comment by prathaptce** [comment]
Sure @prudhvigodithi, I will raise the PR for 1.x as well

**Pull request merged**